dspace-api/src/main/java/org/dspace/handle/HandleServiceImpl.java

```diff
@@ -376,4 +385,38 @@ protected String createId(Context context) throws SQLException {
     public int countTotal(Context context) throws SQLException {
         return handleDAO.countRows(context);
     }
+
+    public String formatHandle(String identifier) {
```

Comment:
Could we add a Unit Test to verify this is working as expected for all of the `IDENTIFIER_PATTERNS` you've defined?  Currently, we seem to only test that it works for "hdl:", and only from the REST API.  We should have some basic Unit tests for this method to ensure it returns the expected string when passed different `identifier` values.

Two other things to note about this method:
* I think it's improperly named. It's not *formatting* a Handle.  It's parsing the handle out of an identifier string. So, I think we should call this `parseHandle()` and describe it as "Parses/extracts the handle prefix/suffix from an identifier string"
* Also this needs an `@Override`